internal/backend/auth/authhttpmiddleware/authmiddleware.go

```diff
@@ -104,3 +105,21 @@ func New(deps Deps) AuthMiddlewareDecorator {
 		return f
 	}
 }
+
+type AuthHeaderExtractor func(*http.Request) sdktypes.User
+
+// just for logs in interceptor
+func AuthorizationHeaderExtractor(deps Deps) AuthHeaderExtractor {
+	return func(r *http.Request) sdktypes.User {
+		if deps.Tokens != nil {
```

Comment:
Go really frowns upon nesting, especially a chain of `if`s. Instead, inverse the conditions, and return `InvalidUser` every time.

See:
- https://go.dev/wiki/CodeReviewComments#indent-error-flow
- https://testing.googleblog.com/2017/06/code-health-reduce-nesting-reduce.html
- https://github.com/uber-go/guide/blob/master/style.md#reduce-nesting

Reply:
I will rewrite it of coz, but I'm strongly disagree here with you.

Code should be clean and facilitate it's understanding and reading. Usually reducing nesting levels is great, but in this case it feels like we are trying to follow the suggestion blindly while forgetting the reasons why go suggests this.

I will explain my point:
In this function there are 2 return paths - happy one and the error one.
In my version (although there are 3 nested ifs, yes), it focuses the reader on understanding the happy path, when it will happen and what are the all conditions needed. So the reader explicitly understands what should be passed to extract user from the header.
In version you suggested - it (1) blows up the code (lines count, so the reader cannot understand at a glance what code is doing) and focuses the user to see the error path, without understanding the happy path (since happy path is everything which is !error ). Thus in order to understand the code  and what should be passed to get a valid user the reader need to go one step further and apply this mental NOT in his head to all error cases.
So in this particular case, mental effort for code understanding is bigger (IMO) in version you suggest.

rewritten